### PR TITLE
Support OIDC core scopes

### PR DESCRIPTION
**Pull request by louischan-oursky**
ref DEV-366

@fungc-io Here is the screenshot of the updated consent screen. The image is annotated with which scope corresponds to which description. However, you actually cannot test this. The OIDC client application you can create in portal is actually a first party OIDC application. A first party application always skip the consent screen.

SCR-20240812-rwwu

### CODE REVIEW

**Comment by fungc-io** [comment]
Ok, how about authUI v2?

**Comment by louischan-oursky** [comment]
> Ok, how about authUI v2?

Regardless of which UI, the consent screen is never shown.

**Comment by fungc-io** [comment]
i see

File: pkg/lib/oauth/oidc/id_token.go [hunk 1]
```diff
@@ -281,30 +260,31 @@ func (ti *IDTokenIssuer) GetUserInfo(userID string, clientLike *oauth.ClientLike
...
+				if ScopeAllowsClaim(scope, k) {
+					isAllowed = true
+					break
+				}
+			}
+
+			if isAllowed {
+				out[k] = v
 			}
 		}
```
Comment (tung2744) [inline pkg/lib/oauth/oidc/id_token.go:285]: What about custom attributes in third party app?

Comment (louischan-oursky) [reply pkg/lib/oauth/oidc/id_token.go:285]: Currently custom attributes are not visible in third party app. I think we need a new feature to make them visible.

The developer should configure which third party application can request the scope to see a custom attribute.

**Review by tung2744 — APPROVED** [review]

**Pull request merged**